### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by maniackcrudelis**
## Problem
- *This app uses master since a long time. Neglecting the testing branch*
- *Global upgrade of the package.*
- *Upgrade to 1.12.1*
- *Add a change_url script*

## Solution
- *Merge master into testing, and now try to merge this testing branch to restore the process.*

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [x] **Code review** : Kay0u
- [x] **Approval (LGTM)** : Kay0u
- [x] **Approval (LGTM)** : JimboJoe
- **CI succeeded** : 
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/rainloop_ynh%20testing%20(Official)/)  
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by JimboJoe** [comment]
Hm... shouldn't we advertise this testing before releasing it...? :thinking:

**Comment by maniackcrudelis** [comment]
If possible, it would be better to merge #48 before. So we would really have something to test with this branch.

**Comment by yalh76** [comment]
> If possible, it would be better to merge #48 before. So we would really have something to test with this branch.

I made a new PR: https://github.com/YunoHost-Apps/rainloop_ynh/pull/52 without merging conflicts

**Comment by alexAubin** [comment]
Idk where to say this but https://github.com/YunoHost-Apps/rainloop_ynh/issues/36 is still relevant ... sounds like we should be installing php-curl somewhere in this app :s

**Comment by maniackcrudelis** [comment]
Waits for https://github.com/YunoHost-Apps/rainloop_ynh/pull/54

**Comment by alexAubin** [comment]
Randomly passing by, made a fresh install on a RPi and everything seems to work smoothly :+1:

File: scripts/install [hunk 1]
```diff
@@ -1,159 +1,201 @@
...
+ynh_add_fpm_config
+
+#=================================================
+# SPECIFIC SETUP
+#=================================================
+# AUTOCONFIG
+#=================================================
+
+
+
```
Comment (kay0u) [inline scripts/install:118]: Too much spaces here

File: check_process [hunk 2]
```diff
@@ -15,23 +15,27 @@
...
     Level 2=auto
     Level 3=auto
-    Level 4=1
-    Level 5=1
-    Level 6=auto
-    Level 7=auto
-    Level 8=0
-    Level 9=0
-    Level 10=0
+	Level 4=1
```
Comment (kay0u) [inline check_process]: Since this PR, check_process have changed, don't need these "level" anymore

**Review by kay0u — COMMENTED** [review]
Really nice work! That was not an easy upgrade :S

**Comment by maniackcrudelis** [comment]
Ready to be reviewed (finally)

**Comment by kay0u** [comment]
We should wait #56

**Comment by maniackcrudelis** [comment]
As #56 add fail2ban support, that means a new week of testing...

**Comment by kay0u** [comment]
As you wish. But is another week for this PR (started in January) really a problem?

**Comment by maniackcrudelis** [comment]
Not really a problem no, but too bad.
Let's see if #56 will be validated soon as well.

**Comment by yalh76** [comment]
> 
> 
> Not really a problem no, but too bad.
> Let's see if #56 will be validated soon as well.

Maybe enough waiting:
- master branch provide rainloop 1.11.3 from september 2017
- testing provide rainloop 1.12.1 from july 2018

And this PR is already the winning app PR: https://dash.yunohost.org/testings ^^

**Comment by yalh76** [comment]
> 
> 
> Not really a problem no, but too bad.
> Let's see if #56 will be validated soon as well.

Maybe enough waiting:
- master branch provide rainloop 1.11.3 from september 2017
- testing provide rainloop 1.12.1 from july 2018

And this PR is already the winning app PR: https://dash.yunohost.org/testings ^^

**Comment by yalh76** [comment]
> 
> 
> Not really a problem no, but too bad.
> Let's see if #56 will be validated soon as well.

Maybe enough waiting:
- master branch provide rainloop 1.11.3 from september 2017
- testing provide rainloop 1.12.1 from july 2018

And this PR is already the winning app PR: https://dash.yunohost.org/testings ^^

File: scripts/change_url [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	ynh_replace_string "$old_domain$old_path" "$new_domain$new_path" "$final_path/index.php"
+elif [ $old_path != "/" ] && [ $new_path = "/" ]
+then
+	ynh_replace_string "$old_domain$old_path" "$new_domain" "$final_path/index.php"
+elif [ $old_path = "/" ] && [ $new_path != "/" ]
+then
+	ynh_replace_string "$old_domain" "$new_domain$new_path" "$final_path/index.php"
+else
+	ynh_die "Error changing the URL"
+fi
```
Comment (kay0u) [inline scripts/change_url]: Hmm something like that :
```bash
ynh_replace_string "$old_domain${old_path%/}" "$new_domain${new_path%/}" "$final_path/index.php"
```

File: scripts/install [hunk 4]
```diff
@@ -1,159 +1,213 @@
...
-	sudo sed -i "s@ALIASTOCHANGE@$final_path@g"  	$final_path/index.php
-	if [ $path = "/" ]; then
-	    sudo sed -i "s@ROOTTOCHANGE@@g"          	$final_path/index.php
-	else
-	    sudo sed -i "s@ROOTTOCHANGE@$path@g"     	$final_path/index.php
-	fi
+cp ../sources/sso/sso.php "$final_path/index.php"
+ynh_replace_string "__DOMAIN__" "$domain" "$final_path/index.php"
+ynh_replace_string "__ALIASTOCHANGE__" "$final_path" "$final_path/index.php"
+if [ $path_url = "/" ]; then
```
Comment (kay0u) [inline scripts/install]: Same here, but less critical:

```bash
ynh_replace_string "__ROOTTOCHANGE__" ${path_url%/}" "$final_path/index.php"
```

File: scripts/upgrade [hunk 5]
```diff
@@ -1,105 +1,161 @@
...
-	sudo sed -i "s@ALIASTOCHANGE@$final_path@g"  	$final_path/index.php
-	if [ $path = "/" ]; then
-	    sudo sed -i "s@ROOTTOCHANGE@@g"          	$final_path/index.php
-	else
-	    sudo sed -i "s@ROOTTOCHANGE@$path@g"     	$final_path/index.php
-	fi
+cp ../sources/sso/sso.php "$final_path/index.php"
+ynh_replace_string "__DOMAIN__" "$domain" "$final_path/index.php"
+ynh_replace_string "__ALIASTOCHANGE__" "$final_path" "$final_path/index.php"
+if [ $path_url = "/" ]; then
```
Comment (kay0u) [inline scripts/upgrade]: And here

**Review by kay0u — COMMENTED** [review]
Except this, everything LGTM

**Review by kay0u — APPROVED** [review]
LGTM & code review \o/

**Review by JimboJoe — APPROVED** [review]
LGTM :+1:

**Comment by JimboJoe** [comment]
Can be merged in 3 days.

**Pull request merged**